tests/compilation_cache_test.py

```diff
@@ -195,6 +199,46 @@ def test_jit(self):
           files_in_directory = len(os.listdir(tmpdir))
           self.assertEqual(files_in_directory, 2)
 
+  @jtu.with_mesh([('x', 2)])
+  def test_pjit(self):
+      with tempfile.TemporaryDirectory() as tmpdir:
+          cc.initialize_cache(tmpdir)
+          @partial(pjit,
+                   in_axis_resources=(P('x'), P('x')),
+                   out_axis_resources=None)
+          def f(x, y):
+              return x + y
+
+          shape = (8, 8)
+          x = np.arange(prod(shape), dtype=np.int64).reshape(shape)
+          f(x, x + 1)
+          files_in_directory = len(os.listdir(tmpdir))
+          self.assertEqual(files_in_directory, 1)
+          x = np.arange(prod(shape), dtype=np.float32).reshape(shape)
+          f(x, x + 1)
+          files_in_directory = len(os.listdir(tmpdir))
+          self.assertEqual(files_in_directory, 2)
+
+  def test_xmap(self):
+      with tempfile.TemporaryDirectory() as tmpdir:
+          cc.initialize_cache(tmpdir)
+          def f(x):
+              return x * 2
+          devices = np.array(jax.local_devices()[:2])
+          if devices.size < 2:
+              raise SkipTest("Test requires 2 devices")
+          x = np.arange(8, dtype=np.int64).reshape((2, 2, 2))
+          with mesh(devices, ('x',)):
```

Comment:
I _think_ you can replace all of this with the `@jtu.with_mesh([('x', 2)])` decorator used on the pjit test. Try it?